pkg/streamingpromql/operators/subquery.go

```diff
@@ -115,9 +126,36 @@ func (s *Subquery) NextStepSamples() (*types.RangeVectorStepData, error) {
 	s.stepData.RangeStart = rangeStart
 	s.stepData.RangeEnd = rangeEnd
 
+	if s.ChildQueryStats != nil && s.ChildQueryStats.TotalSamplesPerStep != nil {
+		temp := s.samplesProcessedInChildQuery(s.stepData.RangeStart, s.stepData.RangeEnd)
+		s.ParentQueryStats.IncrementSamplesAtTimestamp(s.stepData.StepT, temp)
+	}
+
 	return s.stepData, nil
 }
 
+// samplesProcessedInChildQuery returns the number of samples processed in child query that corresponds to the current parent query step.
+func (s *Subquery) samplesProcessedInChildQuery(start, end int64) int64 {
+	if s.ChildQueryStats == nil || s.ChildQueryStats.TotalSamplesPerStep == nil {
+		return 0
+	}
+
+	sum := int64(0)
+	childStartT := s.ChildQueryTimeRange.StartT
+	childInterval := s.ChildQueryTimeRange.IntervalMilliseconds
+	totalChildSteps := len(s.ChildQueryStats.TotalSamplesPerStep)
+	
+	for i := 0; i < totalChildSteps; i++ {
```

Comment:
I think we can be smarter here: we can determine the first value of `i` based on the value of `start`, and determine when to terminate the loop based on the value of `end` (rather than always starting at `i == 0` and advancing until we reach `start`, then breaking when we reach `end`)